src/main/java/simplexity/simplevanish/listeners/TrialSpawnListener.java

```diff
@@ -0,0 +1,35 @@
+package simplexity.simplevanish.listeners;
+
+import org.bukkit.block.BlockState;
+import org.bukkit.block.TrialSpawner;
+import org.bukkit.block.data.BlockData;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.entity.TrialSpawnerSpawnEvent;
+import org.slf4j.Logger;
+import simplexity.simplevanish.SimpleVanish;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@SuppressWarnings("UnstableApiUsage")
+public class TrialSpawnListener implements Listener {
+
+    @EventHandler
+    public void onTrialSpawn(TrialSpawnerSpawnEvent spawnEvent) {
+        TrialSpawner spawner = spawnEvent.getTrialSpawner();
+        List<Player> trackedPlayers = new ArrayList<>(spawner.getTrackedPlayers().stream().toList());
+        List<Player> modifiedPlayers = new ArrayList<>();
+        for (Player player : trackedPlayers) {
+            if (ListenerUtils.userNotVanished(player)) {
+                modifiedPlayers.add(player);
+            }
+        }
+        if (!modifiedPlayers.isEmpty()) return;
```

Comment:
Unnecessary call to ArrayList Constructor, you are already creating a list.

Since you are already calling `stream()`, consider...
```suggestion
        Optional<Player> anyModifiedPlayer = spawner.getTrackedPlayers().stream()
            .filter(player -> ListenerUtils.userNotVanished(player))
            .findAny();
        if (anyModifiedPlayer.isEmpty()) return;
```

Reply:
Fixed